contracts/BIFI/strategies/CompoundV2/StrategyCompoundV2.sol

```diff
@@ -0,0 +1,416 @@
+     * {BORROW_DEPTH_MAX}    - A limit on how many steps we can leverage.
+     * {INTEREST_RATE_MODE}  - The type of borrow debt. Stable: 1, Variable: 2.
+     */
+    uint256 public borrowRate;
+    uint256 public borrowRateMax;
+    uint256 public borrowDepth;
+    uint256 public minLeverage;
+    uint256 constant public BORROW_DEPTH_MAX = 10;
+
+    /**
+     * @dev Helps to differentiate borrowed funds that shouldn't be used in functions like 'deposit()'
+     * as they're required to deleverage correctly.
+     */
+    uint256 public reserves;
+
+    uint256 public balanceOfPool;
+
+    /**
+     * @dev Events that the contract emits
+     */
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+    event StratRebalance(uint256 _borrowRate, uint256 _borrowDepth);
+    event ChargedFees(uint256 callFees, uint256 beefyFees, uint256 strategistFees);
+
+    constructor(
+        uint256 _borrowRate,
+        uint256 _borrowRateMax,
+        uint256 _borrowDepth,
+        uint256 _minLeverage,
+        address[] memory _outputToNativeRoute,
+        address[] memory _outputToWantRoute,
+        address[] memory _markets,
+        address _comptroller,
+        CommonAddresses memory _commonAddresses
+    ) StratFeeManager(_commonAddresses) {
+        borrowRate = _borrowRate;
+        borrowRateMax = _borrowRateMax;
+        borrowDepth = _borrowDepth;
+        minLeverage = _minLeverage;
+
+        iToken = _markets[0];
+        markets = _markets;
+        comptroller = _comptroller;
+        want = IVToken(iToken).underlying();
+
+        output = _outputToNativeRoute[0];
+        native = _outputToNativeRoute[_outputToNativeRoute.length - 1];
+        outputToNativeRoute = _outputToNativeRoute;
+
+        require(_outputToWantRoute[0] == output, "outputToWantRoute[0] != output");
+        require(_outputToWantRoute[_outputToWantRoute.length - 1] == want, "outputToNativeRoute[last] != want");
+        outputToWantRoute = _outputToWantRoute;
+
+        _giveAllowances();
+        IComptroller(comptroller).enterMarkets(markets);
+    }
+
+    // puts the funds to work
+    function deposit() public whenNotPaused {
+        uint256 wantBal = availableWant();
+
+        if (wantBal > 0) {
+            _leverage(wantBal);
+            emit Deposit(balanceOf());
+        }
+    }
+
+    /**
+     * @dev Repeatedly supplies and borrows {want} following the configured {borrowRate} and {borrowDepth}
+     * @param _amount amount of {want} to leverage
+     */
+    function _leverage(uint256 _amount) internal {
+        if (_amount < minLeverage) { return; }
+
+        for (uint i; i < borrowDepth;) {
+            IVToken(iToken).mint(_amount);
+            _amount = _amount * borrowRate / 100;
+            IVToken(iToken).borrow(_amount);
+            unchecked { i++; }
```

Comment:
++i is cheaper

Reply:
Is should not affect the iterator's behavior if placed outside the loop's signature as it's done here, right?